### ISSUE TITLE
Support eclipse-feature to be referenced in target definitions of type "Maven"

### ISSUE THREAD

**Issue by kwin** [eclipse-tycho/tycho#1639]
For certain edge cases it is useful to not only reference bundles/plug-ins via the `location type=Maven` but also artifacts created with packaging `eclipse-feature`.

**Comment by kwin** [comment]
Seems to work indeed, but the dialog is confusing as features don't have an OSGi manifest and there is no dedicated type for it (I know that they use extension .jar, but so do bundle types which have a dedicated entry in the dropdown)
![Target Editor-Maven-Type](https://user-images.githubusercontent.com/185025/200797145-763baf43-aea3-4fe7-92ef-b5b6d5f4d0eb.png)

**Comment by laeubi** [comment]
Yes but first this is actually m2e if/when this should be improved, and second m2e can't know the type of the artifact (this is first determined when resolving the target) and last but not least you can even mix those in one location and those settings apply to all items in the location.

**Comment by kwin** [comment]
The only documentation I found related to target definitions in Tycho was https://wiki.eclipse.org/Tycho/Target_Platform. Would be really good to document the support of Tycho of target definitions including a description of all elements/attributes (particularly Maven dependencies). Also the PDE documentation at https://wiki.eclipse.org/PDE/Target_Definitions and https://help.eclipse.org/latest/index.jsp?topic=/org.eclipse.pde.doc.user/concepts/target.htm are not very verbose and don't mention Maven Dependencies at all.

**Comment by laeubi** [comment]
As said, Maven Target locations are provided my [m2e](https://github.com/eclipse-m2e/m2e-core) not Tycho nor PDE.

**Comment by kwin** [comment]
Can you point me to a documentation stating which packagings are supported by m2e?

**Comment by laeubi** [comment]
You can use any packaging you want if the resulting artifact is:

1. a pom
2. a jar file
